### ISSUE TITLE
False Positive: ProperNullCoalescingOperatorUsageInspection when avoiding missing key notice

### ISSUE THREAD

**Issue by jwwk** [kalessil/phpinspectionsea#1282]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) v3.0.12.2 |
| Language level | PHP 7.1                       |

Current behaviour
--
**Example:**
```php
class IntStorage
{
    /**
     * @var int[]
     */
    private $myArray = [];

    public function getInt(string $key): ?int
    {
        return $this->myArray[$key] ?? null; // warning
    }

    public function setInt(string $key, int $int): void
    {
        $this->myArray[$key] = $int;
    }
}
```
**Warning is displayed:**
*Resolved operands types are not complimentary, while they should be ([int] vs [null]).*
**Inspection Info:** Analyzes null-coalescing operator usage and reports questionable cases.

Expected behaviour
--
Using the null coalescing operator to handle missing keys without a notice is a recommended practice according to the PHP docs:
> In particular, this operator does not emit a notice if the left-hand side value does not exist, just like isset(). This is especially useful on array keys.
[source](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce)

So what I would expect: No warning at all if the type of the expression (`int` or `null`) matches the return type of method (`?int`)

Environment details
---
PhpStorm 2019.1
Build #PS-191.6183.95, built on March 27, 2019

**Comment by orklah** [comment]
I think this has been fixed here:
https://github.com/kalessil/phpinspectionsea/issues/1259

**Comment by jwwk** [comment]
yep, case closed 👍 
You can't grep screenshots 🙈
